### PR TITLE
[4.0] Removing GUI parts of FTP layer

### PR DESCRIPTION
**Pull request by Hackwar**
This removes the GUI parts of the FTP layer. This PR is the counterpart to #31857. The PLT needs to decide if it wants to either merge this or the other PR.

This PR is not entirely complete. There is still FTP code in the joomlaupdate component and the FTP layer has not been removed from the libraries folder.

### CODE REVIEW

File: administrator/components/com_config/src/Controller/ApplicationController.php [hunk 1]
```diff
@@ -82,9 +82,6 @@ public function save()
 
 		$this->app->setUserState('com_config.config.global.data', null);
 
-		// Set FTP credentials, if given.
-		ClientHelper::setCredentialsFromRequest('ftp');
```
Comment (Quy) [inline administrator/components/com_config/src/Controller/ApplicationController.php:86]: Please delete obsolete use statement. Repeat in other files.

File: installation/src/Model/ConfigurationModel.php [hunk 2]
```diff
@@ -503,21 +495,11 @@ public function createConfiguration($options)
 
 		/*
 		 * If the file exists but isn't writable OR if the file doesn't exist and the parent directory
-		 * is not writable we need to use FTP.
+		 * is not writable the user needs to fix this..
```
Comment (Quy) [inline installation/src/Model/ConfigurationModel.php]:
```suggestion
		 * is not writable the user needs to fix this.
```

**Comment by PhilETaylor** [comment]
Is this implementing a "decision" or is this a speculative PR?

**Comment by brianteeman** [comment]
> Is this implementing a "decision" or is this a speculative PR?

As @Hackwar said

> The PLT needs to decide if it wants to either merge this or the other PR.

**Comment by PhilETaylor** [comment]
Stop picking hairs in everything. @Hackwar is ALSO [quoted as saying 4 days ago](https://github.com/joomla/joomla-cms/pull/31857#issuecomment-826089810) that his other PR to FIX the FTP Layer was him volunteering

> to execute decisions the PLT has made

So if the PLT made a decision (unpublished except to him) to FIX the FTP layer why then is this PR here to remove the PR

It is a genuine question that requires answering. 

The last PUBLIC decision made on the FTP Layer that I can find is now almost a year out of date, from @wilsonge, who said the FTP Layer was being kept. Unless that decision has changed, then there is no need for this PR. And if that decision has been changed IT MUST BE MADE PUBLIC and not just shared with a select few people. 

I have now wasted almost a week this month working on fixing issues in the FTP Layer, to remove release blocker for Joomla 4... it would be nice to actually have some leadership announcing their decisions made in private.

**Comment by PhilETaylor** [comment]
I would ask "the PLT Team" directly - however https://volunteers.joomla.org/departments/production-leadership-team seems to not display ANY members and the last Department Joomla 4 report there was from 2017... hahaha

@Hackwar "Hannes Papenberg" is listed here https://volunteers.joomla.org/departments/production as a Department Member, so one would hope that he actually knows what his leadership team decisions are?? Maybe not though. 

I personally emailed Marco Dings many weeks ago - and did not receive a reply on this subject. 

Looking back at my historic emails, I have never received an email reply from him over the years. Always ignored. 

Im doing all I can to move Joomla 4 closer to being released, as the FTP Layer has been repeatedly listed as a Release Blocker...

**Comment by PhilETaylor** [comment]
@Hackwar When is the next scheduled PLT meeting please?

**Comment by Hackwar** [comment]
@PhilETaylor The PLT decided some time ago to keep the FTP layer (I'd have to look through the meeting minutes to find the exact date) Some time after that, I volunteered to fix the installation via FTP and that is the PR that you already participated in. Thanks to your meta issue about all the issues in the FTP layer, I looked into those issues and decided that indeed the FTP layer is out of date and should probably be removed. I brought this up in the PLT almost a week ago (on the 23rd to be exact) and the few people who answered, followed mine (and your) opinion on this.

__However__, at this moment, there is no official decision by the PLT. As long as there is no official decision, the old one stands to keep the FTP layer. I have no idea when that decision will be made.

Whichever way the PLT decides, I want to not waste more time and started this PR. If we decide to keep the FTP layer, my old PR would be there. If we decide to drop it, this PR here could be merged. Yes, both aren't 100% yet, but I'm hoping that we can save at least some time with this work.

The next PLT meeting is Tuesday, May 4th.

Believe me, that I, too, am trying to push J4 closer to release.

**Comment by PhilETaylor** [comment]
Awesome thats. 

> followed mine (and your) opinion on this.

My opinion is just to get whatever needs doing done - but - that needs a new reiteration of the decision to keep or remove the FTP layer and not based on a year old decision - so that no one wastes weeks of time only to be told it is going to be removed. I guess my opinion is that is should be removed as a FTP layer, especially as on investigation its "mainly broken" 

> The next PLT meeting is Tuesday, May 4th.

I will officially suspend all my (published and yet unpublished) work on the FTP Layer pending the May 4th Meeting. 

The only exception to that will be to retain and keep up to date the Summary Issue so that a clearer picture of the issues involved is kept in peoples minds (instead of believing like George did that "it mostly works" even though he admits he has never used or tested it).

Thank you for the clarity your post has given.

**Comment by Hackwar** [comment]
You're welcome. I've asked to come to a decision regarding this earlier than next meeting.

**Comment by Hackwar** [comment]
A motion regarding this is currently being voted on by mail in the PLT.

**Comment by PhilETaylor** [comment]
Thanks for the update. Will the votes cast, by who and which way, be published also?

**Comment by PhilETaylor** [comment]
And is the motion (as I would assume) to Keep the FTP layer or choose to remove it? or does it have some crazy motion text? Is it a binary choice being made?

**Comment by Hackwar** [comment]
I don't think that making the votes public would provide any benefit for any of us, but if you insist on it, I can bring this up in the next meeting.

The motion is open either way. At the end of the voting, we will have a clear answer to either keep or remove the FTP layer. 

To make this clear: This would remove the FTP layer from the CMS. We would keep it in the Filesystem package for now in case someone would still use it, but it would not be available for installation of Joomla or any of the core components filesystem operations. In that regard, this PR would even be done, I think... I've kept my fingers from the com_joomlaupdate component...

**Comment by PhilETaylor** [comment]
Thanks. 

There is not any FTP in the Filesystem FRAMEWORK package (eg: https://github.com/joomla-framework/filesystem/blob/master/src/File.php) 

Some optimisation can be done in the CMS Filesystem package if FTP is no longer going to be part of the CMS, for performance and logic reasons, mainly caching the FTPOptions from ClientHelper so that we dont have to keep going there, checking the config, checking the session, just to return enabled = 0 each time. 

>  I've kept my fingers from the com_joomlaupdate component...

Its not so much the com_joomlaupdate, its the restore.php that's a large issue :) I think Im the only one brave enough in modern history to make large changes to Nic's code ;-) 🤬🤬

The whole class `AKPostprocFTP` could be removed eventually.. but boy will that be a painful day in the office, and getting people to test :)

**Comment by joomdonation** [comment]
> I think I'm the only one brave enough in modern history to make large changes to Nic's code

You might not be the only one. I even tried to debug it to try to fix update using FTP on Windows , lol.

> The whole class AKPostprocFTP could be removed eventually.. but boy will that be a painful day in the office, and getting people to test :)

Guess we could remove **AKPostprocHybrid**, **AKPostprocSFTP**, too. And test that is not hard. We just need to test and make sure update using Direct method works and it will be fine.

**Comment by Hackwar** [comment]
The votes have been cast and the decision is to drop the FTP layer from the system. This means that the CMS does not provide an FTP fallback during installation of the system itself or when writing files in Joomla core components. The code in the CMS Filesystem package will stay in case third party extensions are depending on this. We have not made a decision if that code should now be marked as deprecated or not. Since that would only be possible in Joomla 5.0 anyways, I don't think we have to decide on this now.

With that said, this PR would need to be tested and reviewed.

**Comment by PhilETaylor** [comment]
Please can this be announced officially in some place other than a comment in a GitHub issue buried with all the other issues and comments please... 

Im sure I'll not be the first to say - PHEW. 

Thanks.

**Comment by HLeithner** [comment]
@PhilETaylor it was a production motion "PROD2021/002 PASSED" I expect that this will be published soon.

**Comment by PhilETaylor** [comment]
@HLeithner Thanks - was it a close vote ? hahaaha

**Comment by HLeithner** [comment]
@PhilETaylor Not really only one vote against it.

**Comment by dgrammatiko** [comment]
@HLeithner was there also a decision for @ciar4n 's PR, the Atum repaint?

**Comment by HLeithner** [comment]
> @HLeithner was there also a decision for @ciar4n 's PR, the Atum repaint?

That's a bit off topic here, at least we talked about it and some actions have been done, I would expect more information soon.

**Comment by brianteeman** [comment]
> @PhilETaylor it was a production motion "PROD2021/002 PASSED" I expect that this will be published soon.

Is it off topic to ask if the registry of motions will be updated with all the motions from the last 12 months

**Comment by HLeithner** [comment]
@brianteeman of course it's off topic but I think you know this ;-) I posted your request to production and hope it will be updated soon. ( I can't do more)

**Comment by brianteeman** [comment]
@HLeithner thanks

**Comment by brianteeman** [comment]
@Hackwar submitted a few simple pr to your branch

**Comment by wilsonge** [comment]
> I would ask "the PLT Team" directly - however https://volunteers.joomla.org/departments/production-leadership-team seems to not display ANY members and the last Department Joomla 4 report there was from 2017... hahaha

Just fwiw although Production is behind it's not that bad - not quite sure how the URLs work on the volunteer portal - but this has the more up-to-date reports https://volunteers.joomla.org/departments/production (I guess the other link is the old PLT stuff before Joomla did the restructure)

**Comment by conconnl** [comment]
> @brianteeman of course it's off topic but I think you know this ;-) I posted your request to production and hope it will be updated soon. ( I can't do more)

The registry is located here and updated by Production or Luca and Me, when we receive the information. 
https://www.opensourcematters.org/organisation/team-membership/registry-of-motions/289-2020.html#production-department-motions-2020

Just send the list and it will be updated.

**Comment by Hackwar** [comment]
@brianteeman done

**Comment by brianteeman** [comment]
@conconnl obviously I know that link. Thats why i know it has not been updated for almost a year

@hackwar thanks for merging. I didnt touch anything else as I'm unsure what bits are being kept

**Comment by joomdonation** [comment]
@Hackwar I made a separate PR https://github.com/joomla/joomla-cms/pull/33437 to remove FTP layer from com_joomlaupdate to make it easier for testing (and also save you sometime because I have some experience with that component)

**Comment by Hackwar** [comment]
@joomdonation Thank you.

**Comment by joomdonation** [comment]
@Hackwar Since we do not remove FTPClient, I think the removal of language items from **administrator/language/en-GB/lib_joomla.ini** and **language/en-GB/lib_joomla.ini** should be reverted. Other than that, I think the PR is fine.

**Comment by Hackwar** [comment]
@joomdonation Agreed. I've reverted that.

If we can get 2 positive tests for this then, we can merge this.

**Comment by joomdonation** [comment]
I have tested this item :white_check_mark: successfully on a450cbdb2ee46b381811096893a630a7603dca40<br /><br />- Code review
- Install Joomla
- Save configuration
- Save component options
- Install Extension.
- Install language package
- Create files from template manager

All works as expected. So it is good.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33390">issues.joomla.org/tracker/joomla-cms/33390</a>.</sub>

**Comment by dgrammatiko** [comment]
I have tested this item :white_check_mark: successfully on a450cbdb2ee46b381811096893a630a7603dca40<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33390">issues.joomla.org/tracker/joomla-cms/33390</a>.</sub>

**Comment by joomdonation** [comment]
@richard67 This PR could be set to RTC now.

**Comment by richard67** [comment]
> @richard67 This PR could be set to RTC now.

No need to extra ping me, I am checking every now and then if there are PR's which have 2 tests and would have found it sooner or later.

**Comment by richard67** [comment]
Drone failure is not related to this PR.

**Comment by joomdonation** [comment]
@richard67  OK, will remember that :).

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33390">issues.joomla.org/tracker/joomla-cms/33390</a>.</sub>

File: administrator/components/com_admin/src/Model/SysinfoModel.php [hunk 3]
```diff
@@ -286,7 +286,7 @@ public function &getConfig(): array
 		$registry = new Registry(new \JConfig);
 		$this->config = $registry->toArray();
 		$hidden = [
-			'host', 'user', 'password', 'ftp_user', 'ftp_pass',
+			'host', 'user', 'password',
```
Comment (wilsonge) [inline administrator/components/com_admin/src/Model/SysinfoModel.php]: We shouldn't delete this. configuration.php will not be updated until someone re-saves in global config. Given how broken things are in J3 I agree this is super unlikely but why risk exposing the sensitive properties?

**Review by wilsonge — DISMISSED** [review]

File: administrator/components/com_admin/src/Model/SysinfoModel.php (hunk 3 above)
Comment (Hackwar) [reply administrator/components/com_admin/src/Model/SysinfoModel.php]: fixed

**Comment by richard67** [comment]
~~@wilsonge Does it need again 2 tests after your suggested change has been implemented here? https://github.com/joomla/joomla-cms/pull/33390/commits/f4e45f6f9ea33ba976517a4236d148e3990744e0~~

Just clarified in Glip: No new tests needed. I've restored the previous test results in the issue tracker.

**Comment by richard67** [comment]
I've updated the branch to latest 4.0-dev to get rid of the unrelated javascript-cs error in drone and restored the previous test results in the issue tracker.

**Comment by wilsonge** [comment]
@Hackwar if you can fix the conflict happy to get this in

**Comment by richard67** [comment]
@wilsonge @Hackwar I allowed myself to fix the conflict.

**Comment by Hackwar** [comment]
thx

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**